tools/packaging/common/istio-iptables.sh

```diff
@@ -83,41 +87,39 @@ function isIPv4() {
 #
 # Function return true if agrument is a valid ipv6 address
 #
-function isIPv6() {
+isIPv6() {
   local ipv6section="^[0-9a-fA-F]{1,4}$"
-  addr="$1"
+  addr=$(echo "$1" | sed -e "s/:/:${IFS}/g")
   number_of_parts=0
   number_of_skip=0
-  IFS=':' read -r -a addr <<< "$1"
-  if [ ${#addr[@]} -eq 0 ]; then
+  if [ "$(echo "${addr}" | awk -F"${IFS}"  '{print NF}')" -eq 0 ]; then
      return 1
   fi
-  for part in "${addr[@]}"; do
+  for part in ${addr}; do
+    part=${part%:}
     # check to not exceed number of parts in ipv6 address
-    if [[ ${number_of_parts} -ge 8 ]]; then
+    if [ ${number_of_parts} -ge 8 ]; then
         return 1
     fi
-    if [[ ${number_of_parts} -eq 0 ]] && ! [[ ${part} =~ ${ipv6section} ]]; then
+    if [ ${number_of_parts} -eq 0 ] && ! echo "${part}" | grep -q -E "${ipv6section}"; then
```

Comment:
I am pretty scared by all these changes to iptables. This is one of the most critical code paths for Istio, and it has virtually no code coverage. How confident are we that the script behaves the same now as it did before? Is there anything we can do to get more confidence?

Reply:
> This is one of the most critical code paths for Istio, and it has virtually no code coverage. ...

How come? Sounds pretty scary to me too. If there is good you are afraid to change, that is never a good sign.

But a very good point though. We thought of rewriting `istio-iptables.sh` in go This could address the testing problem and gain confidence. What is your take on that?

Reply:
I am sure there is no *good* reason. In theory, we do have some coverage since all the E2E tests will make use of it. I think long term, a go rewrite is a good idea, we were recently discussing this in another PR that made some changes here.

I am not saying we should not make this change, but we should be cautious about it.

Reply:
Please file an enhancement request to reimplement istio-iptables.sh in go.  We don't need that for 1.2, but it would be fantastic to get rid of bash which is very difficult to unit test.  Code that is difficult to automatically test slows down merge velocity for several reasons.

Reply:
https://github.com/istio/istio/issues/13967 created an issue